### PR TITLE
Support chain certificates in network-conduit-tls

### PR DESCRIPTION
**Pull request by ygale**
This compiles, but it is untested. I believe it ought to work though.

### CODE REVIEW

File: network-conduit-tls/Data/Conduit/Network/TLS.hs [hunk 1]
```diff
@@ -58,19 +60,24 @@ import Control.Monad.Trans.Control
 import Data.Default
 
 
-makeCertDataPath :: FilePath -> FilePath -> TlsCertData
-makeCertDataPath certPath keyPath = TlsCertData (readFile certPath) (readFile keyPath)
-
-makeCertDataBS :: S.ByteString -> S.ByteString -> TlsCertData
-makeCertDataBS certBS keyBS = TlsCertData (return certBS) (return keyBS)
+makeCertDataPath :: FilePath -> [FilePath] -> FilePath -> TlsCertData
```
Comment (snoyberg) [inline network-conduit-tls/Data/Conduit/Network/TLS.hs]: This makes it a breaking change, which I'd rather avoid.

**Comment by ygale** [comment]
Those functions are not exported, so I didn't view that as breaking change. But it is of course possible to create separate versions of them to use in the new smart constructors. Is there a reason why it's important to do that?

**Comment by snoyberg** [comment]
Sorry, my mistake, I thought they were exported. I'll continue my review,
thanks.

**Comment by snoyberg** [comment]
Thanks!

**Pull request merged**